openapi_python_client/templates/property_templates/union_property.py.jinja

```diff
@@ -1,19 +1,31 @@
 {% macro construct(property, source, initial_value=None) %}
-def _parse_{{ property.python_name }}(data: Any) -> {{ property.get_type_string() }}:
-    data = None if isinstance(data, Unset) else data
-    {{ property.python_name }}: {{ property.get_type_string() }}
+def _parse_{{ property.python_name }}(data: object) -> {{ property.get_type_string() }}:
+    {% if "None" in property.get_type_strings_in_union(json=True) %}
+    if data is None:
+        return data
+    {% endif %}
+    {% if "Unset" in property.get_type_strings_in_union(json=True) %}
+    if isinstance(data, Unset):
+        return data
+    {% endif %}
     {% for inner_property in property.inner_properties_with_template() %}
     {% if not loop.last or property.has_properties_without_templates %}
     try:
-    {% from "property_templates/" + inner_property.template import construct %}
+        {{ inner_property.python_name }}: {{ inner_property.get_type_string() }}
+    {% from "property_templates/" + inner_property.template import construct, check_type_for_construct %}
+        if not {{ check_type_for_construct(inner_property, "data") }}:
```

Comment:
In theory we could start doing:

```
if {{ check_type_for_construct(inner_property, "data") }}:
    {{ construct(inner_property, "data", initial_value="UNSET") }}
    return {{ inner_property.python_name }}
```
and avoid the try/except stuff, yeah?

Reply:
I don't think it would work out-of-the-box, because we're also catching non-type errors, like parsing dates (`isoparse("foo")`). Agreed that ideally I think it would be more robust for us to move away from the `try/except` (especially with the generic exception catching).